src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -299,12 +344,16 @@ public virtual bool TryHandleTab()
         {
             if (ExpansionSession != null)
             {
-                var tabbedInsideSnippetField = VSConstants.S_OK == ExpansionSession.GoToNextExpansionField(0);
+                var isFullMethodCallSnippet = !_symbols.IsDefault;
+                var tabbedInsideSnippetField = VSConstants.S_OK == ExpansionSession.GoToNextExpansionField(fCommitIfLast: isFullMethodCallSnippet ? 1 : 0);
 
                 if (!tabbedInsideSnippetField)
                 {
                     ExpansionSession.EndCurrentExpansion(fLeaveCaret: 1);
                     ExpansionSession = null;
+                    _symbols = default;
+                    _symbol = null;
+                    _arguments = _arguments.Clear();
```

Comment:
ick (though i have no bette suggestion).  So this is effectively a very stateful entity that gets reused across snippet invocations?  i would have def loved if we'd originaly written this such that all that statefulness was extracted into someting else that was just cleared out as a single go.

Reply:
Agreed. Was actually planning to wrap the state but didn't get to it yet.

Reply:
⏱️ State is now being wrapped in a `State` object